src/components/ui/Text/Text.tsx

```diff
@@ -7,17 +7,38 @@ import { customClassSwitcher } from '~/core';
 // TODO: Add a core reusable function to check and render an as prop
 
 const COMPONENT_NAME = 'Text';
-
+const componentList = [
+    {
+        label: 'DIV',
+        tag: 'div'
+    },
+    {
+        label: 'SPAN',
+        tag: 'span'
+    },
+    {
+        label: 'P',
+        tag: 'p'
+    },
+    {
+        label: 'LABEL',
+        tag: 'label'
+    },
+]
 export type TextProps = {
     children: React.ReactNode;
     customRootClass?: string;
     className?: string;
+    as?: string;
```

Comment:
You could just limit the type to
```
as?: 'div' | 'span' | 'p' | 'label';
```
And it could be used directly. Especially considering, that key and value are the same

Reply:
\+ you could get rid of the extra `tagMap` with this approach

Reply:
String literal can be also separated (i.e `AvailableTags = 'div' | ...;`)